src/dbcp/archivers/utils.py

```diff
@@ -0,0 +1,105 @@
+"""Utility functions and classes for archivers."""
+
+from abc import ABC, abstractmethod
+
+import google.auth
+import yaml
+from google.cloud import storage
+from pydantic import BaseModel
+
+
+class AbstractArchiver(ABC):
```

Comment:
I hear you that we haven't written other archives so we don't get the benefit of a shared interface. I'm not sure I understand you critique with `self.client` and `self.bucket`. All of the archives will be dumping data to the `dgm-archiver` bucket so I figured it would make sense to setup the connection to the bucket in `AbstractArchiver`.